api/Controllers/RoomController.cs

```diff
@@ -80,9 +81,31 @@ public async Task<IActionResult> GetRooms([FromRoute] int hotelId)
                 .Where(r => r.HotelId == hotelId)
                 .ProjectTo<Room>(_mapper.ConfigurationProvider)
                 .ToArrayAsync();
+            foreach (Room room in rooms)
+            {
+                room.Facilities = await GetRoomFacilitiesAsync(room.Id);
```

Comment:
Why do you fill this property with separate request? Why don't you use navigation properties for many-to-many relation between Hotel and Facility entities?